crates/core_arch/src/aarch64/neon/mod.rs

```diff
@@ -1858,6 +1893,488 @@ pub unsafe fn vcvtq_u32_f32(a: float32x4_t) -> uint32x4_t {
     vcvtq_u32_f32_(a)
 }
 
+/// Shift Left and Insert (immediate)
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(sli, n = 1))]
+#[rustc_args_required_const(2)]
+pub unsafe fn vsli_n_s8(a: int8x8_t, b: int8x8_t, n: i32) -> int8x8_t {
+    if n < 0 || n > 7 {
+        unreachable_unchecked();
```

Comment:
Use a panic here for out-of-bounds values. It's optimized away in practice since this function is going to be inlined.

Reply:
Sounds good, will do :+1: